site/community/bookkeeper_proposals.md

```diff
@@ -0,0 +1,107 @@
+
+### How to make a BP?
+
+Here is the process for making a BP:
+
+1. Create an issue `BP-<number>: [capation of bookkeeper proposal]`. E.g. `BP-1: 64 bits ledger id support`.
+    - Take the next available BP number from this page.
+    - Write a brief description about what BP is for in this issue. This issue will be the master issue for tracking the status of this BP and its implementations.
+      All the implementations of this BP should be listed and linked to this master issues.
+1. Write the proposal for this BP.
+    - Make a copy of the [BP-Template](https://github.com/apache/bookkeeper/tree/master/site/bps/BP-template.md). Name the BP file as `BP-<number>-[caption-of-proposal].md`.
+    ```shell
+    $ cp site/bps/BP-template.md site/bps/BP-xyz-capation-of-proposal.md
+    ```
+    - Fill the sections listed in the BP template.
+        - issue: replace `<issue-number>` with the issue number.
+        - state: "Under Discussion"
+        - release: leave the release to `N/A`. you can only mark a release after a BP is implemented.
+1. Send a PR for this BP. Following the instructions in the pull request template.
+    - add `BP` label to this BP
+    - don't associate this PR with any release or milestone
+1. You can tag committers on this RP for reviewers, or start a `[DISCUSS]` thread on Apache mailing list. If you are sending an email, please make sure that the subject
+   of the thread is of the format `[DISCUSS] BP-<number>: capation of bookkeeper proposal`.
+1. Once the BP is finalized, reviewed and approved by committers, the BP is accepted. The criteria for acceptance is [lazy majority](http://bookkeeper.apache.org/bylaws.html).
+1. Committers merge the PR after a BP is accepted. The development for this BP moves forward with implementations. The BP should be updated if there is anything changed during
+   implementing it.
+1. After all the implementations for a given BP are completed, a new PR should be sent for changing the state of a BP:
+    - state: "Adopted"
+    - release: set to the release that includes this BP.
+1. The final PR for changing BP state will be used as the criteria for marking a BP as completed.
+
+## All Proposals
+
+This section lists all the _bookkeeper proposals_ made to BookKeeper.
+
+*Next Proposal Number: 21*
+
+### Inprogress
+
+Proposal | State
+:--------|:-----
+[BP-4 - BookKeeper Lifecycle Management](https://cwiki.apache.org/confluence/display/BOOKKEEPER/BP-4+-+BookKeeper+Lifecycle+Management) | Draft
+[BP-8 - Queue based auto rereplicator](https://cwiki.apache.org/confluence/display/BOOKKEEPER/BP-8+-+Queue+based+auto+rereplicator) | Draft
+[BP-12 - Improve documentation](https://cwiki.apache.org/confluence/display/BOOKKEEPER/BP-12+-+Improve+documentation) | Accepted
+[BP-14 Relax durability](https://cwiki.apache.org/confluence/display/BOOKKEEPER/BP-14+Relax+durability) | Accepted
```

Comment:
These should be consistent in the link text even if the wiki links themselves are not.